lib/hooks/userspace/hook-koa.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var cls = require('../../cls.js');
+var TraceLabels = require('../../trace-labels.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+var constants = require('../../constants.js');
+var agent;
+
+var SUPPORTED_VERSIONS = '1.x';
+
+function useWrap(use) {
+  return function useTrace() {
+    if (!this._google_trace_patched) {
+      this._google_trace_patched = true;
+      this.use(middleware);
+    }
+    return use.apply(this, arguments);
+  };
+}
+
+function *middleware(next) {
```

Comment:
`function*` as per the guide w/ reasoning: http://www.2ality.com/2014/08/formatting-generator-asterisk.html